src/appleseed/renderer/kernel/intersection/treerepository.h

```diff
@@ -86,8 +86,8 @@ class TreeRepository
 template <typename TreeType>
 TreeRepository<TreeType>::~TreeRepository()
 {
-    for (foundation::each<TreeContainer> i = m_trees; i; ++i)
-        delete i->second.m_tree;
+    for (auto &i : m_trees)
```

Comment:
Whenever possible, when the type is definitely known, I'd prefer that we don't use `auto`.

Reply:
Another thing: `i` means "iterator": it was the correct name when using `each<>` and `const_each<>` but it's no longer the case with range-based for loops. Here, `tree` would be a better name.